### ISSUE TITLE
Support the parameters whose type is `Optional` for annotated services

### ISSUE THREAD

**Issue by trustin** [line/armeria#738]
e.g.

```java
@Get("/users/{userId}")
public HttpResponse getUser(@Param("userId") Optional<String> userId) {
    ....
}
```

... which would let a user choose to use our `@Optional` annotation or JDK's `Optional`.

**Comment by trustin** [comment]
If this is implemented well, we may want to remove our `@Optional` annotation in favor of JDK's `Optional`.

**Comment by hyangtack** [comment]
@trustin 
I tried to implement this issue [here](https://github.com/hyangtack/armeria/commit/8793efe79dc0247654a7721eaa1e45506015f0b2).
But I'm not sure yet whether it's useful or not. A user may specify a default value when he or she uses an optional parameter. In this case, there might have no advantage. For example,

1. Old implementation with `@Optional`
```
        @Get
        @Path("/param/default1")
        public String paramDefault1(RequestContext ctx,
                                    @Param("username") @Optional("hello") String username,
                                    @Param("password") @Optional("world") String password,
                                    @Param("extra") String extra) {
            return username + "/" + password + "/" + extra != null ? extra : "(null)";
        }
```

2. New implementation with Optional<T> and `@Default`
```
        @Get
        @Path("/param/default1")
        public String paramDefault1(RequestContext ctx,
                                    @Param("username") @Default("hello") Optional<String> username,
                                    @Param("password") @Default("world") Optional<String> password,
                                    @Param("extra") Optional<String> extra) {
            return username.get() + "/" + password.get() + "/" + extra.orElse("(null)");
        }
```

To specify a default value clearly, I needed to add `@Default` annotation which is very similar to `@Optional` used before. I think that it might be general use-cases, so the old implementation seems like more clear, from this point of view. What do you think?

**Comment by trustin** [comment]
A user could simply use Optional.getOrElse()?

**Comment by hyangtack** [comment]
@trustin Yes, it's enough on the above example code. But a user may write as follows if the variable uses several times in the codes, so one more unnecessary variable may be required. It's my worry. :-)
```
final String name = username.orElse("someDefaultValue");
```

**Comment by trustin** [comment]
@hyangtack Yes, I think a user can just declare a new local variable if he or she needs to access the parameter more than once.

**Comment by hyangtack** [comment]
@trustin As we discussed on LINE chat, 
- Change `@Optional` to `@Default`
- Support Optional<T> as the same with `@Optional`
Thanks! :-)

**Comment by trustin** [comment]
Fixed via #775
